src/LibraryManager.Vsix/Shared/Telemetry.cs

```diff
@@ -46,5 +48,83 @@ public static void TrackException(string name, Exception exception)
+                List<string> librariesNames_Cancelled = GetLibrariesNames(providerResults.Where(r => r.Cancelled));
+                List<string> librariesNames_Uptodate = GetLibrariesNames(providerResults.Where(r => r.UpToDate));
+
+                if (librariesNames_Success.Count > 0)
+                {
+                    telResult[$"LibrariesCount_{provider}_Success"] = librariesNames_Success.Count;
+                    telResult[$"LibrariesIDs_{provider}_Success"] = new TelemetryPiiProperty(string.Join(" ,", librariesNames_Success));
+                }
+
+                if (librariesNames_Failure.Count > 0)
+                {
+                    telResult[$"LibrariesCount_{provider}_Failure"] = librariesNames_Failure.Count;
+                    telResult[$"LibrariesIDs_{provider}_Failure"] = new TelemetryPiiProperty(string.Join(" ,", librariesNames_Failure));
+                }
+
+                if (librariesNames_Cancelled.Count > 0)
+                {
+                    telResult[$"LibrariesCount_{provider}_Cancelled"] = librariesNames_Cancelled.Count;
+                    telResult[$"LibrariesIDs_{provider}_Cancelled"] = new TelemetryPiiProperty(string.Join(" ,", librariesNames_Cancelled));
+                }
+
+                if (librariesNames_Uptodate.Count > 0)
+                {
+                    telResult[$"LibrariesCount_{provider}_Uptodate"] = librariesNames_Uptodate.Count;
+                    telResult[$"LibrariesIDs_{provider}_Uptodate"] = new TelemetryPiiProperty(string.Join(" ,", librariesNames_Uptodate));
+                }
+
+                foreach (ILibraryOperationResult result in providerResults)
+                {
+                    if (result.InstallationState.Files != null)
+                    {
+                        telResult[$"{provider}_{result.InstallationState.Name}_FilesCount"] = new TelemetryPiiProperty(result.InstallationState.Files.Count);
+                    }
+                }
+            }
+
+            TrackUserTask($@"{operation}_Operation", TelemetryResult.None, telResult.Select(i => new KeyValuePair<string, object>(i.Key, i.Value)).ToArray());
+        }
+
+        private static List<string> GetLibrariesNames(IEnumerable<ILibraryOperationResult> results)
+        {
+            List<string> librariesNames = new List<string>();
+
+            foreach (ILibraryOperationResult result in results)
+            {
+                librariesNames.Add(result.InstallationState.LibraryId);
+            }
+
+            return librariesNames;
+        }
+
+        private static void LogErrors(IEnumerable<ILibraryOperationResult> results)
+        {
+            foreach (ILibraryOperationResult result in results)
+            {
+                foreach (IError error in result.Errors)
+                {
+                    TrackOperation("error", TelemetryResult.Failure, new KeyValuePair<string, object>("code", error.Code));
```

Comment:
We need a better name to recognize what "error". If it is just "error", we don't know what the error it is.